yunwoo/baekjoon/수학/1978-소수-찾기.js

```diff
@@ -0,0 +1,36 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const n = +input[0];
+  const num = input[1].split(" ").map(Number);
+  let count = 0;
+
+  const isPrimeNumberCheck = (x) => {
+    for (let i = 2; i <= Math.sqrt(x); i++) {
+      if (x % i === 0) return false;
+    }
+    return true;
+  };
+
+  for (let i = 0; i < n; i++) {
+    if (num[i] !== 1 && isPrimeNumberCheck(num[i])) count++;
+  }
+
+  console.log(count);
```

Comment:
```suggestion
  const n = +input[0];
  const numArr = input[1].split(" ").map(Number);
  let count = 0;

  const isPrimeNumberCheck = (num) => {
    if (num === 1) return false;
    
    for (let i = 2; i <= Math.sqrt(num); i++) {
      if (num % i === 0) return false;
    }
    return true;
  };

  for (let i = 0; i < n; i++) {
    if (isPrimeNumberCheck(numArr[i])) count++;
  }

  console.log(count);
```
- 깔끔한 풀이 잘봤습니다 🙇‍♂️
- 변수 num보다는 숫자인 배열들이기 때문에 numArr이 더 의미가 확실한 것 같습니다 🙇‍♂️
- isPrimeNumberCheck에서도 마찬가지인데 x보다는 num과 같이 더 명확한 변수명이 좋을 것 같습니다~
- isPrimeNumberCheck안에서 num이 1인 경우에 false를 반환하는 로직도 추가하면 실제 count를 증가시키는 부분에서 간단한 조건식으로 가능할 것 같습니다~!!
- 아래처럼 forEach와 같은 메서드를 사용해서도 깔끔한 풀이가 가능한 것 같습니다~
```js
  const numArr = input[1].split(" ").map(Number);
  let result = 0;

  numArr.forEach((num) => {
    if (isPrimeNumberCheck(num)) result++;
  });

  console.log(result);
```

Reply:
- 앗 맞습니다 네이밍 조금 더 신경쓰겠습니다!!🙆‍♂️
- 아하 함수안에 if문을 추가해주는 방법이 있군요..! 더 깔끔한 것 같습니다!!👍
- 배열의 반복이니 forEach가 더 올바른 방법인거 같습니다!! 좋은 참고가 되었습니다 코멘트 감사합니다 🙇‍♂️👍